--- tdebase/CMakeLists.txt.orig 2012-10-20 13:29:16.000000000 +0400
+++ tdebase/CMakeLists.txt      2012-10-21 04:03:09.000000000 +0400
@@ -219,8 +219,8 @@
 
 if( BUILD_STARTTDE )
   install( PROGRAMS starttde DESTINATION ${BIN_INSTALL_DIR} )
+  install( PROGRAMS migratekde3 r14-xdg-update DESTINATION ${BIN_INSTALL_DIR} )
 endif()
-install( PROGRAMS migratekde3 r14-xdg-update DESTINATION ${BIN_INSTALL_DIR} )
 
 
 ##### write configure files #####################
