--- starttde    2019-01-27 13:01:05.443907841 +0300
+++ starttde.gentoo 2019-01-27 13:01:48.203621371 +0300
@@ -178,7 +178,7 @@

 # Modify the following environment variables only as necessary.

-  if ! is_in_path PATH "$TDEDIR/games" ; then
+#  if ! is_in_path PATH "$TDEDIR/games" ; then
     # Respect the traditional path order. Don't blindly place $TDEDIR/games
     # first in the path. Only place $TDEDIR/games before /usr/games. If packagers
     # are adding $TDEDIR/games elsewhere, then they need to ensure the traditional
@@ -190,9 +190,9 @@
     else
       export PATH=$TDEDIR/games:$PATH
     fi
-  fi
+#  fi

-  if ! is_in_path PATH "$TDEDIR/bin" ]; then
+#  if ! is_in_path PATH "$TDEDIR/bin" ]; then
     # Respect the traditional path order. Don't blindly place $TDEDIR/bin
     # first in the path. Only place $TDEDIR/bin before /usr/bin. This order is
     # consistent with tdelibs/tdesu/stub.cpp. If packagers are adding $TDEDIR/bin
@@ -204,7 +204,7 @@
     else
       export PATH=$TDEDIR/bin:$PATH
     fi
-  fi
+#  fi

   if [ -x /usr/bin/manpath ]; then
     if [ "`manpath 2>/dev/null | grep \"$TDEDIR/share/man\"`" = "" ]; then
