diff --git a/kopete/styles/Makefile.am b/kopete/styles/Makefile.am
index 39e20e7..3d7bcf0 100644
--- a/kopete/styles/Makefile.am
+++ b/kopete/styles/Makefile.am
@@ -1,2 +1,2 @@
-SUBDIRS = Kopete Hacker Clean Clear Konqi Retropete Gaim
+SUBDIRS = Kopete Hacker Clean Clear Konqi Retropete Pidgin
 
diff --git a/kopete/styles/Pidgin/CMakeLists.txt b/kopete/styles/Pidgin/CMakeLists.txt
new file mode 100644
index 0000000..f08e42e
--- /dev/null
+++ b/kopete/styles/Pidgin/CMakeLists.txt
@@ -0,0 +1,12 @@
+#################################################
+#
+#  (C) 2010-2011 Serghei Amelian
+#  serghei (DOT) amelian (AT) gmail.com
+#
+#  Improvements and feedback are welcome
+#
+#  This file is released under GPL >= 2
+#
+#################################################
+
+add_subdirectory( Contents )
diff --git a/kopete/styles/Pidgin/CREDITS b/kopete/styles/Pidgin/CREDITS
new file mode 100644
index 0000000..ee91cc7
--- /dev/null
+++ b/kopete/styles/Pidgin/CREDITS
@@ -0,0 +1,7 @@
+Original textonly Style for Adium written by Mark Fickett
+http://www.adiumxtras.com/index.php?a=xtras&xtra_id=44
+
+Modified justtext Style for Adium written by Huw Rowlands
+http://www.adiumxtras.com/index.php?a=xtras&xtra_id=990
+
+Modified Pidgin Style for Kopete written by Thanos Kyritsis
diff --git a/kopete/styles/Pidgin/Contents/CMakeLists.txt b/kopete/styles/Pidgin/Contents/CMakeLists.txt
new file mode 100644
index 0000000..88724a1
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/CMakeLists.txt
@@ -0,0 +1,12 @@
+#################################################
+#
+#  (C) 2010-2011 Serghei Amelian
+#  serghei (DOT) amelian (AT) gmail.com
+#
+#  Improvements and feedback are welcome
+#
+#  This file is released under GPL >= 2
+#
+#################################################
+
+add_subdirectory( Resources )
diff --git a/kopete/styles/Pidgin/Contents/Info.plist b/kopete/styles/Pidgin/Contents/Info.plist
new file mode 100644
index 0000000..87cd60e
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Info.plist
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" 
+        "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>CFBundleDevelopmentRegion</key>
+	<string>English</string>
+	<key>CFBundleGetInfoString</key>
+	<string>Pidgin Kopete chat Style</string>
+	<key>CFBundleIdentifier</key>
+	<string>Kopete.Pidgin.style</string>
+	<key>CFBundleInfoDictionaryVersion</key>
+	<string>1.0</string>
+	<key>CFBundleName</key>
+	<string>Pidgin</string>
+	<key>CFBundlePackageType</key>
+	<string>KopeteChatStyle</string>
+	<key>DefaultBackgroundColor</key>
+	<string>FFFFFF</string>
+	<key>DefaultFontFamily</key>
+	<string>DejaVu Sans Mono</string>
+	<key>DefaultFontSize</key>
+	<integer>12</integer>
+	<key>DisableCustomBackground</key>
+	<false/>
+	<key>DisplayNameForNoVariant</key>
+	<string>grays</string>
+	<key>ShowsUserIcons</key>
+	<false/>
+</dict>
+</plist>
diff --git a/kopete/styles/Pidgin/Contents/Makefile.am b/kopete/styles/Pidgin/Contents/Makefile.am
new file mode 100644
index 0000000..6940fe8
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Makefile.am
@@ -0,0 +1 @@
+SUBDIRS = Resources
\ No newline at end of file
diff --git a/kopete/styles/Pidgin/Contents/Resources/CMakeLists.txt b/kopete/styles/Pidgin/Contents/Resources/CMakeLists.txt
new file mode 100644
index 0000000..1ab98d5
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/CMakeLists.txt
@@ -0,0 +1,18 @@
+#################################################
+#
+#  (C) 2010-2011 Serghei Amelian
+#  serghei (DOT) amelian (AT) gmail.com
+#
+#  Improvements and feedback are welcome
+#
+#  This file is released under GPL >= 2
+#
+#################################################
+
+add_subdirectory( Incoming )
+add_subdirectory( Outgoing )
+add_subdirectory( Variants )
+
+install( FILES
+    main.css Footer.html Header.html Status.html
+  DESTINATION ${DATA_INSTALL_DIR}/kopete/styles/Pidgin/Contents/Resources )
diff --git a/kopete/styles/Pidgin/Contents/Resources/Footer.html b/kopete/styles/Pidgin/Contents/Resources/Footer.html
new file mode 100644
index 0000000..e69de29
diff --git a/kopete/styles/Pidgin/Contents/Resources/Header.html b/kopete/styles/Pidgin/Contents/Resources/Header.html
new file mode 100644
index 0000000..e69de29
diff --git a/kopete/styles/Pidgin/Contents/Resources/Incoming/Action.html b/kopete/styles/Pidgin/Contents/Resources/Incoming/Action.html
new file mode 100644
index 0000000..0f2993c
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/Incoming/Action.html
@@ -0,0 +1,6 @@
+<div class="KopeteMessage" style="direction: %messageDirection%;">
+    <span style="color:%senderColor%;s"><span class="inActionTime">(%time{%H:%M:%S}%) </span>
+	 <span class="inActionMetacontact">%sender%:&nbsp;</span></span>
+	 <span class="inActionMessage" style="background-color: %textbackgroundcolor{#4386cf}%;">%message%</span>
+</div>
+<div id="insert"></div>
diff --git a/kopete/styles/Pidgin/Contents/Resources/Incoming/CMakeLists.txt b/kopete/styles/Pidgin/Contents/Resources/Incoming/CMakeLists.txt
new file mode 100644
index 0000000..6ff0e0d
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/Incoming/CMakeLists.txt
@@ -0,0 +1,14 @@
+#################################################
+#
+#  (C) 2010-2011 Serghei Amelian
+#  serghei (DOT) amelian (AT) gmail.com
+#
+#  Improvements and feedback are welcome
+#
+#  This file is released under GPL >= 2
+#
+#################################################
+
+install( FILES
+    Action.html Content.html NextContent.html
+  DESTINATION ${DATA_INSTALL_DIR}/kopete/styles/Pidgin/Contents/Resources/Incoming )
diff --git a/kopete/styles/Pidgin/Contents/Resources/Incoming/Content.html b/kopete/styles/Pidgin/Contents/Resources/Incoming/Content.html
new file mode 100644
index 0000000..89d07c9
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/Incoming/Content.html
@@ -0,0 +1,6 @@
+<div class="KopeteMessage" style="direction: %messageDirection%;">
+    <span style="color:%senderColor%;"><span class="inContentTime">(%time{%H:%M:%S}%) </span>
+	 <span class="inMetacontact">%sender%:&nbsp;</span></span>
+	 <span class="IncomingMessage" style="background-color: %textbackgroundcolor{#4386cf}%;">%message%</span>
+</div>
+<div id="insert"></div>
diff --git a/kopete/styles/Pidgin/Contents/Resources/Incoming/Makefile.am b/kopete/styles/Pidgin/Contents/Resources/Incoming/Makefile.am
new file mode 100644
index 0000000..7cd3902
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/Incoming/Makefile.am
@@ -0,0 +1,4 @@
+style_DATA = Action.html Content.html NextContent.html
+styledir = $(kde_datadir)/kopete/styles/Pidgin/Contents/Resources/Incoming
+
+EXTRA_DIST = $(style_DATA)
\ No newline at end of file
diff --git a/kopete/styles/Pidgin/Contents/Resources/Incoming/NextContent.html b/kopete/styles/Pidgin/Contents/Resources/Incoming/NextContent.html
new file mode 100644
index 0000000..89d07c9
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/Incoming/NextContent.html
@@ -0,0 +1,6 @@
+<div class="KopeteMessage" style="direction: %messageDirection%;">
+    <span style="color:%senderColor%;"><span class="inContentTime">(%time{%H:%M:%S}%) </span>
+	 <span class="inMetacontact">%sender%:&nbsp;</span></span>
+	 <span class="IncomingMessage" style="background-color: %textbackgroundcolor{#4386cf}%;">%message%</span>
+</div>
+<div id="insert"></div>
diff --git a/kopete/styles/Pidgin/Contents/Resources/Makefile.am b/kopete/styles/Pidgin/Contents/Resources/Makefile.am
new file mode 100644
index 0000000..f350045
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/Makefile.am
@@ -0,0 +1,5 @@
+SUBDIRS = Incoming Outgoing Variants
+style_DATA = main.css Footer.html Header.html Status.html
+styledir = $(kde_datadir)/kopete/styles/Pidgin/Contents/Resources
+
+EXTRA_DIST = $(style_DATA)
\ No newline at end of file
diff --git a/kopete/styles/Pidgin/Contents/Resources/Outgoing/Action.html b/kopete/styles/Pidgin/Contents/Resources/Outgoing/Action.html
new file mode 100644
index 0000000..a7d6ef0
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/Outgoing/Action.html
@@ -0,0 +1,6 @@
+<div class="KopeteMessage" style="direction: %messageDirection%;">
+    <span style="color:%senderColor%;"><span class="outActionTime">(%time{%H:%M:%S}%) </span>
+	 <span class="outActionMetacontact">%sender%:&nbsp;</span></span>
+	 <span class="outActionMessage" style="background-color: %textbackgroundcolor{#4386cf}%;">%message%</span>
+</div>
+<div id="insert"></div>
diff --git a/kopete/styles/Pidgin/Contents/Resources/Outgoing/CMakeLists.txt b/kopete/styles/Pidgin/Contents/Resources/Outgoing/CMakeLists.txt
new file mode 100644
index 0000000..68bcd9c
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/Outgoing/CMakeLists.txt
@@ -0,0 +1,14 @@
+#################################################
+#
+#  (C) 2010-2011 Serghei Amelian
+#  serghei (DOT) amelian (AT) gmail.com
+#
+#  Improvements and feedback are welcome
+#
+#  This file is released under GPL >= 2
+#
+#################################################
+
+install( FILES
+    Action.html Content.html NextContent.html
+  DESTINATION ${DATA_INSTALL_DIR}/kopete/styles/Pidgin/Contents/Resources/Outgoing )
diff --git a/kopete/styles/Pidgin/Contents/Resources/Outgoing/Content.html b/kopete/styles/Pidgin/Contents/Resources/Outgoing/Content.html
new file mode 100644
index 0000000..9ea2bb6
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/Outgoing/Content.html
@@ -0,0 +1,6 @@
+<div class="KopeteMessage" style="direction: %messageDirection%;">
+    <span style="color:%senderColor%;"><span class="outContentTime">(%time{%H:%M:%S}%) </span>
+    <span class="outMetacontact">%sender%:&nbsp;</span></span>
+	 <span class="OutgoingMessage" style="background-color: %textbackgroundcolor{#4386cf}%;">%message%</span>
+</div>
+<div id="insert"></div>
diff --git a/kopete/styles/Pidgin/Contents/Resources/Outgoing/Makefile.am b/kopete/styles/Pidgin/Contents/Resources/Outgoing/Makefile.am
new file mode 100644
index 0000000..85a558d
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/Outgoing/Makefile.am
@@ -0,0 +1,4 @@
+style_DATA = Action.html Content.html NextContent.html
+styledir = $(kde_datadir)/kopete/styles/Pidgin/Contents/Resources/Outgoing
+
+EXTRA_DIST = $(style_DATA)
diff --git a/kopete/styles/Pidgin/Contents/Resources/Outgoing/NextContent.html b/kopete/styles/Pidgin/Contents/Resources/Outgoing/NextContent.html
new file mode 100644
index 0000000..9ea2bb6
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/Outgoing/NextContent.html
@@ -0,0 +1,6 @@
+<div class="KopeteMessage" style="direction: %messageDirection%;">
+    <span style="color:%senderColor%;"><span class="outContentTime">(%time{%H:%M:%S}%) </span>
+    <span class="outMetacontact">%sender%:&nbsp;</span></span>
+	 <span class="OutgoingMessage" style="background-color: %textbackgroundcolor{#4386cf}%;">%message%</span>
+</div>
+<div id="insert"></div>
diff --git a/kopete/styles/Pidgin/Contents/Resources/Status.html b/kopete/styles/Pidgin/Contents/Resources/Status.html
new file mode 100644
index 0000000..4858785
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/Status.html
@@ -0,0 +1,5 @@
+<div class="KopeteMessage" style="direction: %messageDirection%;">
+    <span class="InternalTime">(%time{%H:%M:%S}%) </span>
+    <span class="InternalDelim">#&nbsp;</span>
+	 <span class="InternalMessage">%message%</span>
+</div>
diff --git a/kopete/styles/Pidgin/Contents/Resources/Variants/CMakeLists.txt b/kopete/styles/Pidgin/Contents/Resources/Variants/CMakeLists.txt
new file mode 100644
index 0000000..4976ef9
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/Variants/CMakeLists.txt
@@ -0,0 +1,14 @@
+#################################################
+#
+#  (C) 2010-2011 Serghei Amelian
+#  serghei (DOT) amelian (AT) gmail.com
+#
+#  Improvements and feedback are welcome
+#
+#  This file is released under GPL >= 2
+#
+#################################################
+
+install( FILES
+    No-Colors.css Name-Colors.css Status-Colors.css Contact-Colors.css
+  DESTINATION ${DATA_INSTALL_DIR}/kopete/styles/Pidgin/Contents/Resources/Variants )
diff --git a/kopete/styles/Pidgin/Contents/Resources/Variants/Contact-Colors.css b/kopete/styles/Pidgin/Contents/Resources/Variants/Contact-Colors.css
new file mode 100644
index 0000000..7a85556
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/Variants/Contact-Colors.css
@@ -0,0 +1,10 @@
+
+@import url(../main.css);
+
+.inContentTime, .inMetacontact {
+	color: inherit;
+}
+
+.outContentTime, .outMetacontact {
+	color: inherit;
+}
\ No newline at end of file
diff --git a/kopete/styles/Pidgin/Contents/Resources/Variants/Makefile.am b/kopete/styles/Pidgin/Contents/Resources/Variants/Makefile.am
new file mode 100644
index 0000000..5bbbc31
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/Variants/Makefile.am
@@ -0,0 +1,4 @@
+style_DATA = No-Colors.css Name-Colors.css Status-Colors.css Contact-Colors.css
+styledir = $(kde_datadir)/kopete/styles/Pidgin/Contents/Resources/Variants
+
+EXTRA_DIST = $(style_DATA)
\ No newline at end of file
diff --git a/kopete/styles/Pidgin/Contents/Resources/Variants/Name-Colors.css b/kopete/styles/Pidgin/Contents/Resources/Variants/Name-Colors.css
new file mode 100644
index 0000000..f9fce97
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/Variants/Name-Colors.css
@@ -0,0 +1,13 @@
+@import url(../main.css);
+
+.inActionTime, .inActionMetacontact, .inActionMessage {
+	color: #000000;
+}
+
+.outActionTime, .outActionMetacontact, .outActionMessage {
+	color: #000000;
+}
+
+.InternalTime, .InternalDelim, .InternalMessage {
+	color: #000000;
+}
\ No newline at end of file
diff --git a/kopete/styles/Pidgin/Contents/Resources/Variants/No-Colors.css b/kopete/styles/Pidgin/Contents/Resources/Variants/No-Colors.css
new file mode 100644
index 0000000..d847a19
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/Variants/No-Colors.css
@@ -0,0 +1,22 @@
+@import url(../main.css);
+
+.inContentTime, .inMetacontact {
+	color: #000000;
+}
+
+.outContentTime, .outMetacontact {
+	color: #000000;
+}
+
+.inActionTime, .inActionMetacontact, .inActionMessage {
+	color: #000000;
+}
+
+.outActionTime, .outActionMetacontact, .outActionMessage {
+	color: #000000;
+}
+
+.InternalTime, .InternalDelim, .InternalMessage {
+	color: #000000;
+}
+
diff --git a/kopete/styles/Pidgin/Contents/Resources/Variants/Status-Colors.css b/kopete/styles/Pidgin/Contents/Resources/Variants/Status-Colors.css
new file mode 100644
index 0000000..373f6d7
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/Variants/Status-Colors.css
@@ -0,0 +1,9 @@
+@import url(../main.css);
+
+.inContentTime, .inMetacontact {
+	color: #000000;
+}
+
+.outContentTime, .outMetacontact {
+	color: #000000;
+}
\ No newline at end of file
diff --git a/kopete/styles/Pidgin/Contents/Resources/main.css b/kopete/styles/Pidgin/Contents/Resources/main.css
new file mode 100644
index 0000000..393d2f1
--- /dev/null
+++ b/kopete/styles/Pidgin/Contents/Resources/main.css
@@ -0,0 +1,40 @@
+/* textonly by Mark Fickett, 2004. Poke/edit/maul, but leave credit - think GPL */
+/* Generally: naib.webhop.org Adium-related: naib.webhop.org/~markfickett/adium */
+
+body {
+	background: #ffffff;
+}
+
+div {
+	overflow: auto;
+}
+
+a:link { text-decoration: none; }
+a:visited { text-decoration: none; }
+a:hover { text-decoration: underline; }
+a:active { text-decoration: none; }
+
+.inMetacontact, .outMetacontact, .inActionMetacontact, .outActionMetacontact, .InternalDelim {
+	font-weight: bold;
+}
+
+.inContentTime, .inMetacontact {
+	color: #a82f2f;
+}
+
+.outContentTime, .outMetacontact {
+	color: #16569e;
+}
+
+.inActionTime, .inActionMetacontact, .inActionMessage {
+	color: green;
+}
+
+.outActionTime, .outActionMetacontact, .outActionMessage {
+	color: green;
+}
+
+.InternalTime, .InternalDelim, .InternalMessage {
+	color: #9400d3;
+}
+
diff --git a/kopete/styles/Pidgin/Makefile.am b/kopete/styles/Pidgin/Makefile.am
new file mode 100644
index 0000000..331c9b5
--- /dev/null
+++ b/kopete/styles/Pidgin/Makefile.am
@@ -0,0 +1 @@
+SUBDIRS = Contents
\ No newline at end of file
