diff --git a/kio/kio/kservice.cpp b/kio/kio/kservice.cpp
index 4964103..368fe9a 100644
--- a/kio/kio/kservice.cpp
+++ b/kio/kio/kservice.cpp
@@ -198,7 +198,7 @@ KService::init( KDesktopFile *config )
 
   m_strExec = config->readPathEntry( "Exec" );
   if (kde4application && !m_strExec.startsWith("/")) {
-    m_strExec = "KDEHOME=$HOME/.trinity XDG_DATA_DIRS=/usr/share KDEDIRS=/usr/ XDG_CONFIG_DIRS=/etc/xdg/ PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:$PATH "+m_strExec;
+    m_strExec = "KDEHOME=$HOME/.kde4 XDG_DATA_DIRS=/usr/share KDEDIRS=/usr/ XDG_CONFIG_DIRS=/etc/xdg/ PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:$PATH "+m_strExec;
   } else if (config->readBoolEntry("X-KDE-SubstituteUID")) {
     int space = m_strExec.find(" ");
     if (space==-1)
