diff --git a/translations/qt_zh-tw.ts b/translations/qt_zh-tw.ts
index 8260fe4..0f0eaa8 100644
--- a/translations/qt_zh-tw.ts
+++ b/translations/qt_zh-tw.ts
@@ -115,7 +115,7 @@
     </message>
     <message>
         <source>Menu</source>
-        <translation>Menu</translation>
+        <translation>選單</translation>
     </message>
     <message>
         <source>Help</source>
@@ -337,7 +337,7 @@
     </message>
     <message>
         <source>A&amp;lpha channel:</source>
-        <translation>Alpha 色頻(&amp;L)：</translation>
+        <translation>透明色頻(&amp;L)：</translation>
     </message>
     <message>
         <source>&amp;Basic colors</source>
@@ -429,11 +429,11 @@
     </message>
     <message>
         <source>Read: %1</source>
-        <translation>讀取： %1</translation>
+        <translation>讀取：%1</translation>
     </message>
     <message>
         <source>Write: %1</source>
-        <translation>寫入： %1</translation>
+        <translation>寫入：%1</translation>
     </message>
     <message>
         <source>Cancel</source>
@@ -1194,7 +1194,7 @@ to
     </message>
     <message>
         <source>Customize...</source>
-        <translation>自訂...</translation>
+        <translation>自訂…</translation>
     </message>
 </context>
 <context>
@@ -1264,7 +1264,7 @@ to
     </message>
     <message>
         <source>&lt;h3&gt;About Qt&lt;/h3&gt;&lt;p&gt;This program uses Qt version %1.&lt;/p&gt;&lt;p&gt;Qt is a C++ toolkit for multiplatform GUI &amp;amp; application development.&lt;/p&gt;&lt;p&gt;Qt provides single-source portability across MS&amp;nbsp;Windows, Mac&amp;nbsp;OS&amp;nbsp;X, Linux, and all major commercial Unix variants.&lt;br&gt;Qt is also available for embedded devices.&lt;/p&gt;&lt;p&gt;Qt is a Trolltech product. See &lt;tt&gt;http://www.trolltech.com/qt/&lt;/tt&gt; for more information.&lt;/p&gt;</source>
-        <translation>&lt;h3&gt;關於 Qt&lt;/h3&gt;&lt;p&gt;本程式使用 Qt 版本 %1。&lt;/p&gt;&lt;p&gt;Qt 是用於多平台 GUI 和應用程式開發的 C++ 工具套件。&lt;/p&gt;&lt;p&gt;Qt 為 MS&amp;nbsp;Windows、Mac&amp;nbsp;OS&amp;nbsp;X、Linux 和所有主要的商用 Unix 變異版本，提供跨平台的單一原始碼可攜性。&lt;br&gt;Qt 也可用於內嵌裝置。&lt;/p&gt;&lt;p&gt;Qt 是 Trolltech 的產品。如需詳細資訊，請參閱 &lt;tt&gt;http://www.trolltech.com/qt/&lt;/tt&gt;。&lt;/p&gt;</translation>
+        <translation>&lt;h3&gt;關於 Qt&lt;/h3&gt;&lt;p&gt;本程式使用 Qt 版本 %1。&lt;/p&gt;&lt;p&gt;Qt 是用於多平臺 GUI 和應用程式開發的 C++ 工具套件。&lt;/p&gt;&lt;p&gt;Qt 為 MS&amp;nbsp;Windows、Mac&amp;nbsp;OS&amp;nbsp;X、Linux 和所有主要的商用 Unix 變異版本，提供跨平臺的單一原始碼可攜性。&lt;br&gt;Qt 也可用於內嵌裝置。&lt;/p&gt;&lt;p&gt;Qt 是 Trolltech 的產品。如需詳細資訊，請參閱 &lt;tt&gt;http://www.trolltech.com/qt/&lt;/tt&gt;。&lt;/p&gt;</translation>
     </message>
 </context>
 <context>
@@ -1282,7 +1282,7 @@ to
     </message>
     <message>
         <source>Aliases: %1</source>
-        <translation>別名： %1</translation>
+        <translation>別名：%1</translation>
     </message>
     <message>
         <source>unknown</source>
@@ -1338,7 +1338,7 @@ to
     </message>
     <message>
         <source>Browse...</source>
-        <translation>瀏覽...</translation>
+        <translation>瀏覽…</translation>
     </message>
     <message>
         <source>Options</source>
@@ -1386,123 +1386,123 @@ to
     </message>
     <message>
         <source>A0 (841 x 1189 mm)</source>
-        <translation>A0 (841 x 1189 mm)</translation>
+        <translation>A0 (841 x 1189 公釐)</translation>
     </message>
     <message>
         <source>A1 (594 x 841 mm)</source>
-        <translation>A1 (594 x 841 mm)</translation>
+        <translation>A1 (594 x 841 公釐)</translation>
     </message>
     <message>
         <source>A2 (420 x 594 mm)</source>
-        <translation>A2 (420 x 594 mm)</translation>
+        <translation>A2 (420 x 594 公釐)</translation>
     </message>
     <message>
         <source>A3 (297 x 420 mm)</source>
-        <translation>A3 (297 x 420 mm)</translation>
+        <translation>A3 (297 x 420 公釐)</translation>
     </message>
     <message>
         <source>A4 (210x297 mm, 8.26x11.7 inches)</source>
-        <translation>A4 (210x297 mm，8.26x11.7 英吋)</translation>
+        <translation>A4 (210x297 公釐，8.26x11.7 英吋)</translation>
     </message>
     <message>
         <source>A5 (148 x 210 mm)</source>
-        <translation>A5 (148 x 210 mm)</translation>
+        <translation>A5 (148 x 210 公釐)</translation>
     </message>
     <message>
         <source>A6 (105 x 148 mm)</source>
-        <translation>A6 (105 x 148 mm)</translation>
+        <translation>A6 (105 x 148 公釐)</translation>
     </message>
     <message>
         <source>A7 (74 x 105 mm)</source>
-        <translation>A7 (74 x 105 mm)</translation>
+        <translation>A7 (74 x 105 公釐)</translation>
     </message>
     <message>
         <source>A8 (52 x 74 mm)</source>
-        <translation>A8 (52 x 74 mm)</translation>
+        <translation>A8 (52 x 74 公釐)</translation>
     </message>
     <message>
         <source>A9 (37 x 52 mm)</source>
-        <translation>A9 (37 x 52 mm)</translation>
+        <translation>A9 (37 x 52 公釐)</translation>
     </message>
     <message>
         <source>B0 (1000 x 1414 mm)</source>
-        <translation>B0 (1000 x 1414 mm)</translation>
+        <translation>B0 (1000 x 1414 公釐)</translation>
     </message>
     <message>
         <source>B1 (707 x 1000 mm)</source>
-        <translation>B1 (707 x 1000 mm)</translation>
+        <translation>B1 (707 x 1000 公釐)</translation>
     </message>
     <message>
         <source>B2 (500 x 707 mm)</source>
-        <translation>B2 (500 x 707 mm)</translation>
+        <translation>B2 (500 x 707 公釐)</translation>
     </message>
     <message>
         <source>B3 (353 x 500 mm)</source>
-        <translation>B3 (353 x 500 mm)</translation>
+        <translation>B3 (353 x 500 公釐)</translation>
     </message>
     <message>
         <source>B4 (250 x 353 mm)</source>
-        <translation>B4 (250 x 353 mm)</translation>
+        <translation>B4 (250 x 353 公釐)</translation>
     </message>
     <message>
         <source>B5 (176 x 250 mm, 6.93x9.84 inches)</source>
-        <translation>B5 (176 x 250 mm，6.93x9.84 英吋)</translation>
+        <translation>B5 (176 x 250 公釐，6.93x9.84 英吋)</translation>
     </message>
     <message>
         <source>B6 (125 x 176 mm)</source>
-        <translation>B6 (125 x 176 mm)</translation>
+        <translation>B6 (125 x 176 公釐)</translation>
     </message>
     <message>
         <source>B7 (88 x 125 mm)</source>
-        <translation>B7 (88 x 125 mm)</translation>
+        <translation>B7 (88 x 125 公釐)</translation>
     </message>
     <message>
         <source>B8 (62 x 88 mm)</source>
-        <translation>B8 (62 x 88 mm)</translation>
+        <translation>B8 (62 x 88 公釐)</translation>
     </message>
     <message>
         <source>B9 (44 x 62 mm)</source>
-        <translation>B9 (44 x 62 mm)</translation>
+        <translation>B9 (44 x 62 公釐)</translation>
     </message>
     <message>
         <source>B10 (31 x 44 mm)</source>
-        <translation>B10 (31 x 44 mm)</translation>
+        <translation>B10 (31 x 44 公釐)</translation>
     </message>
     <message>
         <source>C5E (163 x 229 mm)</source>
-        <translation>C5E (163 x 229 mm)</translation>
+        <translation>C5E (163 x 229 公釐)</translation>
     </message>
     <message>
         <source>DLE (110 x 220 mm)</source>
-        <translation>DLE (110 x 220 mm)</translation>
+        <translation>DLE (110 x 220 公釐)</translation>
     </message>
     <message>
         <source>Executive (7.5x10 inches, 191x254 mm)</source>
-        <translation>Executive (7.5x10 英吋，191x254 mm)</translation>
+        <translation>Executive (7.5x10 英吋，191x254 公釐)</translation>
     </message>
     <message>
         <source>Folio (210 x 330 mm)</source>
-        <translation>Folio (210 x 330 mm)</translation>
+        <translation>Folio (210 x 330 公釐)</translation>
     </message>
     <message>
         <source>Ledger (432 x 279 mm)</source>
-        <translation>Ledger (432 x 279 mm)</translation>
+        <translation>Ledger (432 x 279 公釐)</translation>
     </message>
     <message>
         <source>Legal (8.5x14 inches, 216x356 mm)</source>
-        <translation>Legal (8.5x14 英吋，216x356 mm)</translation>
+        <translation>Legal (8.5x14 英吋，216x356 公釐)</translation>
     </message>
     <message>
         <source>Letter (8.5x11 inches, 216x279 mm)</source>
-        <translation>Letter (8.5x11 英吋，216x279 mm)</translation>
+        <translation>Letter (8.5x11 英吋，216x279 公釐)</translation>
     </message>
     <message>
         <source>Tabloid (279 x 432 mm)</source>
-        <translation>Tabloid (279 x 432 mm)</translation>
+        <translation>Tabloid (279 x 432 公釐)</translation>
     </message>
     <message>
         <source>US Common #10 Envelope (105 x 241 mm)</source>
-        <translation>美國一般 10 號信封 (105 x 241 mm)</translation>
+        <translation>美國一般 10 號信封 (105 x 241 公釐)</translation>
     </message>
     <message>
         <source>Setup Printer</source>
@@ -1691,7 +1691,7 @@ to
     <name>QToolBar</name>
     <message>
         <source>More...</source>
-        <translation>其他...</translation>
+        <translation>其他…</translation>
     </message>
 </context>
 <context>
@@ -1811,7 +1811,7 @@ to
     </message>
     <message>
         <source>&amp;Help</source>
-        <translation>說明(&amp;H)</translation>
+        <translation>求助(&amp;H)</translation>
     </message>
 </context>
 <context>
@@ -1988,7 +1988,7 @@ to
     </message>
     <message>
         <source>Paste special...</source>
-        <translation>選擇性貼上...</translation>
+        <translation>選擇性貼上…</translation>
     </message>
     <message>
         <source>Clear</source>
